lemur/tests/test_messaging.py

```diff
@@ -225,6 +225,30 @@ def test_send_authority_expiration_notifications():
     assert send_authority_expiration_notifications() == (2, 0)
 
 
+@mock_ses
+def test_send_expiring_deployed_certificate_notifications():
```

Comment:
I don't particularly like how I've written this test, but I also haven't come up with a better way to do it. At the moment, it makes an actual network call to these domains to check for their current certificates. In order to get the notification to actually send, I need to make the code believe a cert in the DB matches a cert deployed somewhere - unless I were to mock the TLS cert download logic, which I'm not sure how to do in this code base.

I could also remove the test, though I don't love that option either.

Suggestions welcome.

Reply:
yeah, you are right, the actual network call might raise some eyebrows.

would it be worthwhile to mock the server?
https://dev.to/bowmanjd/two-methods-for-testing-https-api-calls-with-python-and-pytest-and-also-communicating-with-the-in-laws-388n

and serve a test certificates?
https://github.com/Netflix/lemur/blob/3783fbeaa1645bbee022827f4f53ffb12dd65a61/lemur/tests/vectors.py